File: examples/c/mega_flow/executor.c
#include <stdlib.h>

int execute(const char *cmd) {
    /* SINK — system() · c.cmdi.system · CWE-78 */
    return system(cmd);
}

int clean_twin(void) {
    /* NEGATIVE — same sink kind with a constant argument must not report. */
    return system("echo clean");
}

File: examples/go/mega_flow/executor.go
package main

import execpkg "os/exec"

// Execute — SINK, os/exec.Command run through /bin/sh -c.
func Execute(cmd string) int {
	// go.cmdi.exec_command · severity=critical · CWE-78
	_ = execpkg.Command("/bin/sh", "-c", cmd).Run()
	return 0
}

func cleanTwin() int {
	// NEGATIVE — same sink kind with a constant argument must not report.
	_ = execpkg.Command("/bin/sh", "-c", "echo clean").Run()
	return 0
}
